server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1860,9 +1860,14 @@ private CompactionStats _majorCompact(MajorCompactionReason reason) throws IOExc
 
       if (inputFiles.isEmpty()) {
         if (reason == MajorCompactionReason.USER) {
-          // no work to do
-          lastCompactID = compactionId.getFirst();
-          updateCompactionID = true;
+          if (compactionId.getSecond().getIterators().isEmpty()) {
+            log.debug("No-op major compaction by USER on 0 input files because no iterators present.");
+            lastCompactID = compactionId.getFirst();
+            updateCompactionID = true;
+          } else {
+            log.debug("Major compaction by USER on 0 input files with iterators.");
+            filesToCompact = new HashMap<FileRef,DataFileValue>();
```

Comment:
No, you're actually right. It's best to keep changes as concise as you can (don't change a bunch of other things in one JIRA issue). It helps people go back in the future and understand what changes were made in relation to the JIRA issue and related conversation.

I don't recall if we've had discussion on the subject of Java7 features aside than "use the new features in new code you write as you'd like." While we support 1.5 and 1.6, if we change 1.7 to use diamond everywhere, it's likely to just cause more of a headache in merges than it's worth in the codebase. That's just my $0.02 though. Feel free to poll the dev list or make a JIRA issue about it though. Others might have different opinions :)